### PR TITLE
feat: add aria selected to nav listitem

### PR DESCRIPTION
**Pull request by csiyang**
# Description

Add aria-selected to navigationListItem

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/29147486/todos/5302095064)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Tests pass

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged into main

### CODE REVIEW

File: apps/journeys-admin/src/components/PageWrapper/NavigationDrawer/NavigationListItem/NavigationListItem.spec.tsx [hunk 1]
```diff
@@ -43,4 +44,15 @@ describe('NavigationListItem', () => {
     fireEvent.click(getByRole('button', { name: 'menu item' }))
     expect(onClick).toHaveBeenCalled()
   })
+
+  it('should be selected', () => {
```
Comment (csiyang) [inline apps/journeys-admin/src/components/PageWrapper/NavigationDrawer/NavigationListItem/NavigationListItem.spec.tsx:48]: This covers all the selection tests that was previously in NavigationDrawer.spec

File: apps/journeys-admin/src/components/PageWrapper/NavigationDrawer/NavigationListItem/NavigationListItem.spec.tsx [hunk 2]
```diff
@@ -8,11 +8,12 @@ describe('NavigationListItem', () => {
...
         icon={<AbcRoundedIcon />}
         label="menu item"
-        selected
+        selected={false}
       />
     )
-    expect(getByTestId('AbcRoundedIcon')).toBeInTheDocument()
-    expect(getByTestId('AbcRoundedIcon')).toHaveStyle(`color: "#fff"`)
+    expect(
+      getByTestId('AbcRoundedIcon').parentElement?.parentElement
```
Comment (storyworks) [inline apps/journeys-admin/src/components/PageWrapper/NavigationDrawer/NavigationListItem/NavigationListItem.spec.tsx:15]: Get by role :)

**Review by storyworks — APPROVED** [review]

File: apps/journeys-admin/src/components/PageWrapper/NavigationDrawer/NavigationDrawer.spec.tsx [hunk 3]
```diff
@@ -51,7 +51,9 @@ describe('NavigationDrawer', () => {
         </FlagsProvider>
       </MockedProvider>
     )
-    expect(getByTestId('ShopRoundedIcon')).toHaveStyle(` color: '#fff'`)
+    expect(
+      getByTestId('ShopRoundedIcon').parentElement?.parentElement
```
Comment (storyworks) [inline apps/journeys-admin/src/components/PageWrapper/NavigationDrawer/NavigationDrawer.spec.tsx:55]: getByRole

**Review by storyworks — APPROVED** [review]

**Comment by storyworks** [comment]
Might change aria-selected of button since it's not a conventional attribute to add to button https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

**Comment by storyworks** [comment]
Also later need to update this use select ListItemButton `selected` prop to get the color change from the theme.

**Pull request merged**